src/common/main.lua

```diff
@@ -69,11 +70,92 @@ main.init = function()
 			["Message-Id"] = msg.Id,
 		})
 		-- Send HyperBEAM patch message with updated ACL that has the ant removed
+		-- only send if the acl is not empty, as we will nuke the entire ACL from the patch device if it is empty
+		if #utils.keys(acl) > 0 then
+			ao.send({
+				device = "patch@1.0",
+				cache = { acl = acl },
+			})
+		end
+	end)
 
-		ao.send({
-			device = "patch@1.0",
-			cache = { acl = acl },
-		})
+	utils.createActionHandler(ActionMap.BatchUnregister, function(msg)
+		assert(msg.From == Owner, "Only ANT Registry owner can batch unregister")
+		local antIds = json.decode(msg.Data)
+		assert(type(antIds) == "table" and #antIds > 0, "msg.Data must be a table of antIds, recieved " .. type(antIds))
+
+		-- dedupe and validate the antIds
+		local antIdsToUnregister = {}
+		--- At the end of the loop, if there are any errors, we will send a notice to the owner with the error
+		--- messages for each antId that had an error.
+		local antIdErrorMap = {}
```

Comment:
suggestion: instead of sending a notice, the return object can include `removed` and `failed` with a list of each.